### ISSUE TITLE
Regression issue: Can't edit entries in array of slug

### ISSUE THREAD

**Issue by alexbchr** [sanity-io/sanity#2379]
**Describe the bug**

When using an `array` of `slug` fields, the individual slugs are not editable (when creating them or editing them). It seems like the dialog does not open and there is no error in console. I even tried with other `editModal` options without any success. This was previously working when I was using version 2.1.6 of Sanity.

**To Reproduce**

1. Have a document with the following field:
```tsx
{
    title: 'Array of Slugs',
    name: 'slugs_array',
    type: 'array',
    of: [{ type: 'slug' }],
}
```

2. Go add a new entry in the array (the dialog does not show up). If you click or double-click the new entry, it is not editable.

**Expected behavior**

When adding a new slug in the array, it should open the edit dialog. When clicking an entry in the array, it should open the dialog to edit the entry.

**Which versions of Sanity are you using?**

```
@sanity/cli             2.6.0 (up to date)
@sanity/base            2.6.0 (up to date)
@sanity/cli             2.6.0 (up to date)
@sanity/components      2.2.6 (up to date)
@sanity/core            2.6.1 (up to date)
@sanity/default-layout  2.6.0 (up to date)
@sanity/default-login   2.2.6 (up to date)
@sanity/desk-tool       2.6.1 (up to date)
@sanity/vision          2.2.6 (up to date)
```

**What operating system are you using?**

Mac OS Big Sur 11.2.1

**Browser**

Chrome 88.0.4324.192

**Which versions of Node.js / npm are you running?**

Run `npm -v && node -v` in the terminal and copy-paste the result here.

```
6.14.8
v12.20.0
```

**Comment by bjornwang** [comment]
Same here.

**Comment by thobas-dnvgl** [comment]
Just noticed this issue on our Sanity Studio too. I had to revert to **version 2.1.x** of `@sanity/%%%` packages to have make it work again.

**Comment by thobas-dnvgl** [comment]
v2.2.0 was the introduction of the new Sanity UI library to Sanity Studio for primitive field types: `string`, `text`, `number`, `boolean`, and `slug`. Coincidence?

**Comment by rexxars** [comment]
Fixed in [v2.7.0](https://github.com/sanity-io/sanity/releases/tag/v2.7.0) - thanks for reporting!
